front/pages/api/w/[wId]/vaults/[vId]/members.ts

```diff
@@ -1,85 +1,58 @@
 
 import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
 import type { Authenticator } from "@app/lib/auth";
-import { GroupResource } from "@app/lib/resources/group_resource";
 import { UserResource } from "@app/lib/resources/user_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
 import { apiError } from "@app/logger/withlogging";
 
-export type GetGroupResponseBody = {
-  group: GroupType & {
-    members: UserType[];
-  };
-};
-
-export type PatchGroupResponseBody = {
-  group: GroupType;
-};
+export interface PatchVaultMembersResponseBody {
+  vault: VaultType;
+}
 
 async function handler(
   req: NextApiRequest,
-  res: NextApiResponse<
-    WithAPIErrorResponse<GetGroupResponseBody | PatchGroupResponseBody>
-  >,
+  res: NextApiResponse<WithAPIErrorResponse<PatchVaultMembersResponseBody>>,
   auth: Authenticator
 ): Promise<void> {
-  const { gId } = req.query;
-
-  if (typeof gId !== "string") {
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+  if (!vault || !vault.canWrite(auth)) {
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "group_not_found",
-        message: "The group you requested was not found.",
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
       },
     });
   }
 
-  const groupRes = await GroupResource.fetchById(auth, gId);
-  // Check if the user has access to the group to get members list
-  if (groupRes.isErr()) {
+  if (!vault.isRegular()) {
     return apiError(req, res, {
-      status_code: 404,
+      status_code: 400,
       api_error: {
-        type: "group_not_found",
-        message: "The group you requested was not found.",
+        type: "invalid_request_error",
+        message: "Only regular vaults can have members.",
       },
     });
   }
 
   switch (req.method) {
-    case "GET":
-      const members = await groupRes.value.getActiveMembers(auth);
-      return res.status(200).json({
-        group: {
-          ...groupRes.value.toJSON(),
-          members: members.map((member) => member.toJSON()),
-        },
-      });
-    case "PATCH":
-      if (!auth.isAdmin()) {
-        // Only admins can patch
-        return apiError(req, res, {
-          status_code: 403,
-          api_error: {
-            type: "workspace_auth_error",
-            message: "Only users that are `admins` can administrate groups.",
-          },
-        });
-      }
-      if (groupRes.value.isSystem()) {
-        return apiError(req, res, {
-          status_code: 403,
-          api_error: {
-            type: "invalid_request_error",
-            message: "System group can't be updated.",
-          },
-        });
-      }
+    case "PATCH": {
+      const regularGroups = vault.groups.filter(
+        (group) => group.kind === "regular"
+      );
+      // Assert that there is exactly one regular group associated with the vault
```

Comment:
I wonder if it isn't better to do the "regular" group selection in the front end (weak opinion)

Why: while it's nice to hide the complexity of groupS for now, it will not stand the test of time, i'm not sure of the value of hiding instead of just embracing it.

note: what happened to the "GET" endpoint ?

Reply:
I killed the group GET endpoint for now since it's not used anymore. We will expose the group entity once we offer users provisioning but as of now, we want to invite users to think in term of "vaults/spaces" rather than groups.

Reply:
> since it's not used anymore

From a previous PR then, right ?

Reply:
This PR, it was only used in the UI to add/remove members on a vault.